core/errors/error-match-error.ts

```diff
@@ -2,7 +2,7 @@ import { MatchError } from "./match-error";
 
 export class ErrorMatchError extends MatchError {
 
-   public constructor(actualError: Error,
+   public constructor(actualError: Error | undefined | null,
```

Comment:
I tend to try not to uses casts if there is another way. I'm not sure what you mean by "Should this be the union type everywhere" as the compiler will be insisting this union type will need to be used "everywhere". Reading about [advanced types](https://www.typescriptlang.org/docs/handbook/advanced-types.html) we could create an Interface or a Type Alias. Type Alias seems a possibility, maybe something like one of these or maybe you've got something else in mind:
```
type ErrorOrUndefOrNull = Error | undefined | null;
type ErrorOrUndefNull = Error | undefined | null;
type ErrorUndefNull = Error | undefined | null;
```